### ISSUE TITLE
Jets startup time is really long in local development with docker on MacOS

### ISSUE THREAD

**Issue by onnimonni** [tongueroo/jets#39]
I'm seeing quite high waiting times (1m17s) before anything happens.

I'm running the latest jets gem (0.8.17) with ruby 2.5.1 and nodejs v10.11.0.

I shared my docker setup here: https://github.com/onnimonni/jets-project
Similar setup has been working really well for my rails projects.

These are the commands I used inside the docker container:
```bash
$ time jets generate scaffold Post title:string
      invoke  active_record
      create    db/migrate/20180922134209_create_posts.rb
      create    /models/post.rb
      invoke  resource_route
       route    resources :posts
      invoke  scaffold_controller
      create    /controllers/posts_controller.rb
      invoke    erb
      create      /views/posts
      create      /views/posts/index.html.erb
      create      /views/posts/edit.html.erb
      create      /views/posts/show.html.erb
      create      /views/posts/new.html.erb
      create      /views/posts/_form.html.erb
      invoke    helper
      create      /helpers/posts_helper.rb

real	1m17.385s
user	0m1.030s
sys	0m0.150s
$ time jets db:create
FATAL:  password authentication failed for user "dbuser"
Couldn't create database for {"adapter"=>"postgresql", "encoding"=>"utf8", "pool"=>"5", "database"=>"development", "username"=>"dbuser", "password"=>"dbpass", "host"=>"db"}
jets aborted!
PG::ConnectionBad: FATAL:  password authentication failed for user "dbuser"
...

Tasks: TOP => db:create
(See full trace by running task with --trace)

real	1m17.327s
user	0m0.920s
sys	0m0.230s
```

**Comment by tongueroo** [comment]
Thanks for the report. Took a gander at the Jets example project you’ve set up. Thanks for putting it together.  It helps. 👍

This likely has to do with the fact that Jets uses Dir.glob a lot to do inflection and figure out what it needs to do. So Jets is reading a bunch of files on the filesystem. The performance issue then surfaces with Docker shared volumes on a macosx host. Looks like you’ve got a macosx host and are mounting a folder for the app code https://github.com/onnimonni/jets-project/blob/master/docker-compose.yml#L8 Have ran into slowness with Docker, macosx and shared volumes many times before 😞 Even ended up writing a rsync tool https://github.com/tongueroo/docksync to workaround the issue. Gut tells me that is the issue right off the bat though. 

Understand that the need the shared volume so that we can edit files locally and see changes immediately reflected on the Docker host. For a test to confirm the theory though,  can you try removing the shared volumes and testing the performance speed without the shared volumes? Would like to confirm my gut thoughts. 

Would like to improve the performance here to be closer to Rails. Curious to see what happens. Have seen that as any app gets larger, the Docker volume on macosx becomes a problem because it takes time to load the app files themselves. There is no noticeable performance overhead with shared volumes when the host is Linux, only with MacOSX. Also have tried using Docker with NFS volumes instead and that helps but it wasn’t enough. 

Will tag this will help wanted.

**Comment by pocheptsov** [comment]
I've tried @onnimonni project setup on Debian Linux and got this:
```
docker run -it -v $(pwd)/src:/app -e AWS_REGION=eu-west-1 jetsproject_server bash

> export TEST=true
> time jets generate scaffold TestPost title:string
      invoke  active_record
      create    db/migrate/20180924045646_create_test_posts.rb
      create    /models/test_post.rb
      invoke  resource_route
       route    resources :test_posts
      invoke  scaffold_controller
      create    /controllers/test_posts_controller.rb
      invoke    erb
      create      /views/test_posts
      create      /views/test_posts/index.html.erb
      create      /views/test_posts/edit.html.erb
      create      /views/test_posts/show.html.erb
      create      /views/test_posts/new.html.erb
      create      /views/test_posts/_form.html.erb
      invoke    helper
      create      /helpers/test_posts_helper.rb

real    0m1.719s
user    0m1.120s
sys     0m0.330s
```
@tongueroo can we use `Jets::Timing` to help recording startup issue?

**Comment by onnimonni** [comment]
Thanks for helping out! The test from @pocheptsov seems to confirm that this is just caused from slow bind mounts in MacOS.

My problem is that I have too many ruby projects with different version dependencies so using rbenv or locally installed ruby gets too time consuming. Also docker helps others to start working with the same project faster because the environments are well defined.

I guess the biggest problem is the `node_modules` folder which is bind mounted to MacOS.
There isn't that many files elsewhere but that folder contains so many files.

I will try this again with setup where those are moved away from the project folder.

**Comment by tongueroo** [comment]
Released in v0.8.18: [CHANGELOG](https://github.com/tongueroo/jets/blob/master/CHANGELOG.md)  Thanks again for the helpful reporting.

**Comment by onnimonni** [comment]
Excellent! and thanks for the fast fix!
